### PR TITLE
JS, MainmenuSearch: #31567, add JS separately

### PR DESCRIPTION
**Pull request by nhaagen**
Moved the JS parts from template to separate file; fixes https://mantis.ilias.de/view.php?id=31567

### CODE REVIEW

**Comment by Amstutz** [comment]
Sooo much love for that one.

File: Services/Search/classes/class.ilMainMenuSearchGUI.php [hunk 1]
```diff
@@ -54,13 +54,12 @@ protected function initRefIdFromQuery() : void
...
     }
 
-
-
     public function getHTML() : string
     {
         iljQueryUtil::initjQuery();
         iljQueryUtil::initjQueryUI();
         $this->tpl = new ilTemplate('tpl.main_menu_search.html', true, true, 'Services/Search');
+        $tpl->addJavascript('Services/Search/js/SearchMainMenu.js');
```
Comment (chfsx) [inline Services/Search/classes/class.ilMainMenuSearchGUI.php]: isn't `$tpl` undefined here?

Comment (nhaagen) [reply Services/Search/classes/class.ilMainMenuSearchGUI.php]: you're right

**Comment by Amstutz** [comment]
@smeyer-ilias : could you look into this, the error shows up constantly in the js console and seems very annoying. Thx.

**Comment by alex40724** [comment]
@smeyer-ilias I merged this one, however the background request fails with a ilCtrlException: ilCtrl::callBaseClass was not given a baseclass and the request doesn&#039;t include one either. in file /usr/local/var/www/ilias/Services/UICore/classes/class.ilCtrl.php on line 139

I will follow up in the Mantis report.

**Pull request merged**